h2o-docs/src/product/migrating.rst

```diff
@@ -1,9 +1,39 @@
 .. _migration:
 
-Migrating to H2O 3
-==================
+Migration Guide
+===============
 
-H2O 3 offers a lot of improvements over H2O 2, including:
+Migrating Between H2O-3 versions
+--------------------------------
+
+H2O-3 does its best to keep backwards compatibility between major versions, but sometimes breaking changes are needed in order to improve code quality and to address issues.
+
+Please refer to the following sections if you encounter any API-related issue when upgrading H2O-3.
+
+**Note**: H2O-3 versions follow the naming convention ``3.major.minor.patch``. 
+
+From 3.22 or Below to 3.24
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Java API
+''''''''
+
+The following classes were moved and/or renamed:
+
+=================================================   ======================================
+  Until 3.22                                          From 3.24
+=================================================   ======================================
+``hex.StackedEnsembleModel``                        ``hex.ensemble.StackedEnsembleModel``
+``hex.StackedEnsembleModel.MetalearnerAlgorithm``   ``hex.ensemble.Metalearner.Algorithm``
+``ai.h2o.automl.AutoML.algo``                       ``ai.h2o.automl.Algo``
+=================================================   ======================================
+
+Some internal methods of ``StackedEnsemble`` and ``StackedEnsembleModel`` are no longer public, but this should not impact anyone.
+
+Migrating to H2O-3
```

Comment:
@angela0xdata can we remove the h2o-2 to h2o-3 section, it is really not relevant anymore